src/util/file_reader.js

```diff
@@ -21,20 +21,38 @@ class NewlineReader {
      * in memory.
      * @param {nb.NativeFSContext} fs_context 
      * @param {string} filepath 
-     * @param {'EXCLUSIVE' | 'SHARED'} [lock]
-     */
-    constructor(fs_context, filepath, lock) {
+     * @param {{
+     *  lock?: 'EXCLUSIVE' | 'SHARED'
+     *  bufsize?: number;
+     *  return_leftover?: boolean;
+     *  warn_too_long?: boolean;
+     * }} [cfg]
+     **/
+    constructor(fs_context, filepath, cfg) {
         this.path = filepath;
-        this.lock = lock;
-        this.buf = Buffer.alloc(4096);
+        this.lock = cfg?.lock;
+        this.return_leftover = Boolean(cfg?.return_leftover);
+        this.warn_too_long = Boolean(cfg?.warn_too_long);
```

Comment:
I think this should be renamed to "skip_overflow_lines" with default false, but will be set to true for the persistent logger use case.